### PR TITLE
Support `target="_blank"` for `<router-link>`

### PR DESCRIPTION
**Pull request by fnlctrl**

### CODE REVIEW

File: src/components/link.js [hunk 1]
```diff
@@ -49,6 +49,11 @@ export default {
         // don't redirect on right click
         /* istanbul ignore if */
         if (e.button !== 0) return
+        // don't redirect if `target="_blank"`
+        /* istanbul ignore if */
+        const target = this.$el && this.$el.attributes.getNamedItem('target')
+        if (target && target.value === '_blank') return
```
Comment (phanan) [inline src/components/link.js]: Two things:
- Why not `const target = this.$el && this.$el.getAttribute('target')`?
- Won't this fail for `target="_blank "` (with an extra space)?

**Review by phanan — COMMENTED** [review]
I don't really know how to use this GitHub review. Should have made it a comment instead :P

File: src/components/link.js (hunk 1 above)
Comment (fnlctrl) [reply src/components/link.js]: Fixed, I guess I'm just too unfamiliar with native dom api :joy:

File: src/components/link.js [hunk 2]
```diff
@@ -49,6 +49,11 @@ export default {
         // don't redirect on right click
         /* istanbul ignore if */
         if (e.button !== 0) return
+        // don't redirect if `target="_blank"`
+        /* istanbul ignore if */
+        const target = this.$el && this.$el.getAttribute('target')
+        if (target && target.toLowerCase().split(' ').indexOf('_blank') > -1) return
```
Comment (hogart) [inline src/components/link.js]: Since code is written in ES Net, shouldn't it be `target.toLowerCase().split(' ').includes('_blank')`?

Comment (fnlctrl) [reply src/components/link.js]: That cannot be transpiled, and would require a polyfill.

**Review by posva — APPROVED** [review]

**Comment by posva** [comment]
@fnlctrl I absolutely forgot about this, sorry

**Comment by fnlctrl** [comment]
@posva Totally forgot this too 😂😂😂
I think I'll just wait for Evan to get back to this.

File: src/components/link.js [hunk 3]
```diff
@@ -49,6 +49,11 @@ export default {
         // don't redirect on right click
         /* istanbul ignore if */
         if (e.button !== 0) return
+        // don't redirect if `target="_blank"`
+        /* istanbul ignore if */
+        const target = this.$el && this.$el.getAttribute('target')
```
Comment (yyx990803) [inline src/components/link.js]: We should use `e.target` here. `this.$el` may be a wrapping `<li>` if `tag` prop is used.

File: src/components/link.js (hunk 2 above)
Comment (yyx990803) [reply src/components/link.js]: how about just `/\b_blank\b/i.test(target)`

**Review by yyx990803 — CHANGES_REQUESTED** [review]

**Comment by bedlate** [comment]
@yyx990803 , hello you da, but how can I do it in the instance, like this:

`this.$router.push({name: 'users.index', target: '_blank'})`

**Comment by Mouvedia** [comment]
If it's supported you should update [this line](https://github.com/vuejs/vuejs.org/commit/489145a05fc875660f1d5e0e57946629eeea8a8a#diff-2ff218682023cc310e3acd06a7271e63R270).

**Pull request merged**